File: src/main/java/com/example/dto/item/ItemDto.java
package com.example.dto.item;

import java.util.UUID;
import com.example.dto.category.CategoryDto;
import com.example.dto.image.ImageDto;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.AllArgsConstructor;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ItemDto {
    private UUID uuid;
    private String name;
    private String description;
    private Double price;
    private CategoryDto category;
    private ImageDto image;
}

File: src/main/java/com/example/dto/item/ItemDto.kt
package com.example.dto.item

import com.example.dto.category.CategoryDto
import com.example.dto.image.ImageDto
import org.springframework.data.redis.core.RedisHash
import java.util.UUID

@RedisHash("items")
data class ItemDto(
    val uuid: UUID,
    val name: String,
    val description: String,
    val price: Double,
    val category: CategoryDto,
    val image: ImageDto?,
)
